R/datasets.R

```diff
@@ -0,0 +1,627 @@
+
+#' Register a Dataset in the workspace
+#'
+#' @description
+#' Register the Dataset in the workspace, making it available to other users of the workspace.
+#'
+#' @param workspace The AzureML workspace in which the Dataset is to be registered.
+#' @param name The name of the Dataset in the workspace.
+#' @param description A description of the Dataset.
+#' @param tags Named list of tags to give the Dataset. Defaults to NULL.
+#' @param create_new_version Boolean to register the dataset as a new version under the specified name.
+#' @return The registered Dataset object.
+#' @export
+#' @md
+register_dataset <- function(dataset, workspace, name, description = NULL,
+                             tags = NULL, create_new_version = FALSE) {
+  dataset$register(workspace, name, description, tags, create_new_version)
+}
+
+#' Unregister all versions under the registration name of this dataset from the workspace.
+#'
+#' @description
+#' Unregister all versions under the registration name of this dataset from the workspace.
+#'
+#' @param dataset The dataset to be unregistered.
+#' @return None
+#' @export
+#' @md
+unregister_all_dataset_versions <- function(dataset) {
+  dataset$unregister_all_versions()
+}
+
+#' Get a registered Dataset from the workspace by its registration name.
+#'
+#' @description
+#' Get a registered Dataset from the workspace by its registration name.
+#'
+#' @param dataset The Dataset object.
+#' @param workspace The existing AzureML workspace in which the Dataset was registered.
+#' @param name The registration name.
+#' @param version The registration version. Defaults to "latest".
+#' @return The registered Dataset object.
+#' @export
+#' @md
+get_dataset_by_name <- function(workspace, name, version = "latest") {
+  azureml$data$abstract_dataset$AbstractDataset$get_by_name(workspace,
+                                                            name,
+                                                            version)
+}
+
+#' Get Dataset by ID.
+#'
+#' @description
+#' Get a Dataset which is saved to the workspace using its ID.
+#'
+#' @param workspace The existing AzureML workspace in which the Dataset is saved.
+#' @param id The ID of the dataset
+#' @return The Dataset object
+#' @export
+#' @md
+get_dataset_by_id <- function(workspace, id) {
+  azureml$data$abstract_dataset$AbstractDataset$get_by_id(workspace, id)
+}
+
+#' Return the named list for input datasets.
+#'
+#' @description
+#' Return the named list for input datasets.
+#'
+#' @param name The name of the input dataset
+#' @param run The run taking the dataset as input
+#' @return A dictionary with the name being the dataset's name and value being
+#' the delivered data. If the mode is set to mount or download, it will return
+#' the base path of the delivered data. If the mode is set to direct, it will
+#' return the dataset object.
+#' @export
+#' @md
+get_input_dataset_from_run <- function(name, run) {
+  run$input_datasets(name)
```

Comment:
this method does not take any parameters. It returns a dictionary from which you need to extract the dataset object by passing its name as the key.

Reply:
yeah, should be `run$input_datasets[name]`

Reply:
also, can you set `run = NULL` and then inside the method:
```
if (is.null(run)) {
    run <- get_current_run()
  }
```
similar to what we do for the log_* methods (e.g. `log_metric_to_run`: https://github.com/Azure/azureml-sdk-for-r/blob/master/R/run.R)